crawl-ref/source/artefact.cc

```diff
@@ -303,6 +303,11 @@ bool is_special_unrandom_artefact(const item_def &item)
            && (_seekunrandart(item)->flags & UNRAND_FLAG_SPECIAL);
 }
 
+bool can_generate(const item_def &item)
+{
+    return !(get_unrand_entry(item.unrand_idx)->flags & UNRAND_FLAG_NOGEN);
+}
+
```

Comment:
Now that I think of it, this has been obsoleted by `archaeologist_unrands()` which already checks for the NOGEN flag, so we can remove this.